internal/controlplane/handlers_profile.go

```diff
@@ -578,6 +578,54 @@ func (s *Server) GetProfileStatusByProject(ctx context.Context,
 	return res, nil
 }
 
+// PatchProfile updates a profile for a project with a partial request
+func (s *Server) PatchProfile(ctx context.Context, ppr *minderv1.PatchProfileRequest) (*minderv1.PatchProfileResponse, error) {
+	patch := ppr.GetPatch()
+	entityCtx := engine.EntityFromContext(ctx)
+
+	err := entityCtx.Validate(ctx, s.store)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "error in entity context: %v", err)
+	}
+
+	if ppr.GetId() == "" {
+		return nil, util.UserVisibleError(codes.InvalidArgument, "profile ID must be specified")
+	}
+
+	profileID, err := uuid.Parse(ppr.GetId())
+	if err != nil {
+		return nil, util.UserVisibleError(codes.InvalidArgument, "Malformed UUID")
+	}
+
+	params := db.UpdateProfileParams{ID: profileID}
```

Comment:
where do we set the `project` for the given profile? Seems to me like we should add it to the `UpdateProfileParams` to ensure that we only update the profile if it belongs to the project in the context.